apps/web/src/server/lib/nunjucks-template-builders/summary-list-builder.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * @param {any} header
+ * @param {any[]} rowArray
+ * @returns {object}
+ */
+export function generateSummaryList(header, rowArray) {
+	const rows = [];
+	for (const row of rowArray) {
+		rows.push({
+			key: {
+				text: row.title
+			},
+			value: {
+				html: formatRowValue(row)
+			},
+			actions: {
+				items: [
+					{
+						href: row.actionLink,
+						text: row.actionText,
+						visuallyHiddenText: row.title
+					}
+				]
+			}
+		});
+	}
+	return {
+		card: {
+			title: {
+				text: header
+			}
+		},
+		rows: rows
+	};
+}
+
+/**
+ * @param {{ valueType: string; value: string; }} row
+ */
+function formatRowValue(row) {
+	let rowValueAsHtml = '';
+	if (Array.isArray(row.value)) {
+		const arrayOfValues = row.value;
+		for (const value of arrayOfValues) {
+			const htmlTags = getHtmlTags(row.valueType, value);
+			rowValueAsHtml += `${htmlTags[0]}${value}${htmlTags[1]}<br>`;
+		}
+	} else {
+		const htmlTags = getHtmlTags(row.valueType, row.value);
+		rowValueAsHtml = `${htmlTags[0]}${row.value}${htmlTags[1]}`;
+	}
+	return rowValueAsHtml;
+}
+
+/**
+ * @param {string} type text | link
+ * @param {string} link optional: if type is link
+ * @returns {[string, string]} <{{openingTag}}>, </{{closingTag}}>
+ */
+function getHtmlTags(type, link) {
```

Comment:
Would this be better returning an object of {start, end}? Will make the code that uses it clearer I think.